format/Message.fbs

```diff
@@ -117,6 +117,40 @@ table DictionaryBatch {
   isDelta: bool = false;
 }
 
+/// A range of field nodes, identified by their offset in the schema.
+/// The offsets are zero-indexed.
+struct FieldNodeRange {
+  /// The starting offset (inclusive)
+  start: long;
+
+  /// The ending offset (exclusive)
+  end: long;
+}
+
+/// A data header describing the shared memory layout of a "bag" of "columns".
+/// It is similar to a RecordBatch but not every top level FieldNode is required
+/// to be included in the wire payload.
```

Comment:
Thinking ahead - how do the APIs for this look like? In Java, this would be a "ragged" VectorSchemaRoot?

Reply:
Understandable. In C++ we would need a container similar to RecordBatch, and I suppose in Java something similar.